### PR TITLE
Environment brdf fix

### PR DESCRIPTION
**Pull request by raytranuk**
This PR:
- Re-introduces an environment BRDF for physically based IBL with fixes to problems that caused it to be reverted in https://github.com/playcanvas/engine/pull/3141
- Adjusts environment map pre-filtering to use GGX distribution

Before this PR:
image
With this PR:
image

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: examples/graphics/material-anisotropic.html [hunk 1]
```diff
@@ -76,6 +76,7 @@
 
         var createSphere = function (x, y, z) {
             var material = new pc.StandardMaterial();
+            material.diffuse = new pc.Color(1,1,1);
```
Comment (willeastcott) [inline examples/graphics/material-anisotropic.html:79]:
```suggestion
material.diffuse = new pc.Color(1, 1, 1);
```

Comment (raytranuk) [reply examples/graphics/material-anisotropic.html:79]: thanks Will

**Comment by mvaligursky** [comment]
I looks good to me as far as I can tell.
I assume visual engine tests behave as expected?

**Comment by willeastcott** [comment]
Have you tried running Seemore with this PR? https://playcanvas.com/project/612100/overview/seemore

**Comment by raytranuk** [comment]
> I looks good to me as far as I can tell.
> I assume visual engine tests behave as expected?

Thanks - yes I ran the visual engine tests, and they did come back with some expected differences - which I feel look more correct than previously

**Comment by raytranuk** [comment]
> Have you tried running Seemore with this PR? https://playcanvas.com/project/612100/overview/seemore

I haven't - but will run it and post some comparison screens here shortly

File: src/graphics/program-lib/chunks/reproject.frag [hunk 2]
```diff
@@ -231,6 +231,17 @@ vec3 hemisphereSamplePhong(vec2 uv, float specPow) {
     return vec3(cos(phi) * sinTheta, sin(phi) * sinTheta, cosTheta);
 }
 
+vec3 hemisphereSampleGGX(vec2 uv, float specPow) {
+    float roughness = 1.0-(log2(specPow)/11.0);
```
[the remark this answers is not in the document]
Comment (raytranuk) [reply src/graphics/program-lib/chunks/reproject.frag:235]: thanks @slimbuck!

**Review by mvaligursky — APPROVED** [review]

**Pull request merged**